lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -511,6 +517,32 @@ instance FromText PoolId where
         where
             textDecodingError = Left . TextDecodingError . show
 
+-- | Coins are stored as Lovelace (reminder: 1 Lovelace = 1e-6 ADA)
+newtype Stake = Stake { getStake :: Word64 } deriving (Show, Eq, Generic)
```

Comment:
What about `Quantity "lovelace" Word64` to be consistent with other amounts (and have JSON instances for free..)

Reply:
fixed